src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -170,13 +81,33 @@ abstract class BaseIR {
   }
 }
 
+object MatrixLocalValue {
+  def apply(sampleIds: IndexedSeq[Annotation]): MatrixLocalValue =
+    MatrixLocalValue(Annotation.empty,
+      sampleIds,
+      Annotation.emptyIndexedSeq(sampleIds.length))
+}
+
+case class MatrixLocalValue(
+  globalAnnotation: Annotation,
+  sampleIds: IndexedSeq[Annotation],
+  sampleAnnotations: IndexedSeq[Annotation]) {
+  assert(sampleIds.length == sampleAnnotations.length)
+
+  def nSamples: Int = sampleIds.length
+
+  def dropSamples(): MatrixLocalValue = MatrixLocalValue(globalAnnotation,
```

Comment:
is there a forthcoming rename that will `s/samples/columns/`?

Reply:
Yes, when we hit a quiescent point, maybe after 0.2 beta?